src/EntryPoint.sol

```diff
@@ -488,9 +413,18 @@ contract EntryPoint is
         returns (uint256 gUsed, bytes4 err)
     {
         UserOp calldata u = _extractUserOp(encodedUserOp);
+
         uint256 g = Math.coalesce(uint96(combinedGasOverride), u.combinedGas);
         uint256 gStart = gasleft();
 
+        if (
+            u.prePaymentMaxAmount > u.totalPaymentMaxAmount
+                || u.prePaymentAmount > u.prePaymentMaxAmount
+                || u.totalPaymentAmount > u.totalPaymentMaxAmount
+        ) {
+            err = PaymentError.selector;
+        }
```

Comment:
low hanging fruit
```solidity
LibBit.or(u.prePaymentAmount > u.prePaymentMaxAmount, 
    LibBit.or(
        u.prePaymentMaxAmount > u.totalPaymentMaxAmount, 
        u.totalPaymentAmount > u.totalPaymentMaxAmount
    )
)
```

i might add a three argument `or` to `LibBit` just for aesthetics.